.github/workflows/macos.yml

```diff
@@ -0,0 +1,43 @@
+# Workflow for macOS (latest).
+name: macOS
+
+on:
+  push:
+    branches:
+      - "**"
+    paths-ignore:
+      - '**.md'
+  pull_request:
+    branches:
+      - "main"
+    paths-ignore:
+      - '**.md'
+
+jobs:
+  build-and-test:
+    name: Build and Test
+    runs-on: macos-latest
+    # We want to run on external PRs, but not on our own internal PRs as
+    # they'll be run by the push to the branch
+    if: github.event_name == 'push' || github.event.pull_request.head.repo.full_name != '3rdparty/stout-flags'
```

Comment:
What's an "external PR" vs an "internal PR"? Would we not need this if we just had the `push` and not the `pull_request` part above?